From 078d811a71edb89fa626882dea86b8fbf059b40f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fernando=20Tarl=C3=A1=20Cardoso=20Lemos?= <fernandotcl@gmail.com>
Date: Fri, 27 Aug 2010 20:54:41 -0300
Subject: [PATCH] Validate the ws number for client assignment.

---
 src/cfgparse.y |   15 ++++++++++-----
 1 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/src/cfgparse.y b/src/cfgparse.y
index 234e8fa..3ba788a 100644
--- a/src/cfgparse.y
+++ b/src/cfgparse.y
@@ -493,13 +493,18 @@ workspace_name:
 assign:
         TOKASSIGN WHITESPACE window_class WHITESPACE optional_arrow assign_target
         {
-                printf("assignment of %s\n", $<string>3);
+                DLOG("assignment of %s\n", $<string>3);
 
                 struct Assignment *new = $<assignment>6;
-                printf("  to %d\n", new->workspace);
-                printf("  floating = %d\n", new->floating);
-                new->windowclass_title = $<string>3;
-                TAILQ_INSERT_TAIL(&assignments, new, assignments);
+                if (new->floating != ASSIGN_FLOATING_ONLY && new->workspace < 1) {
+                        DLOG("Invalid client assignment, workspace number %d out of range\n", new->workspace);
+                        free(new);
+                } else {
+                        DLOG("  to %d\n", new->workspace);
+                        DLOG("  floating = %d\n", new->floating);
+                        new->windowclass_title = $<string>3;
+                        TAILQ_INSERT_TAIL(&assignments, new, assignments);
+                }
         }
         ;
 
-- 
1.7.1

