diff --git a/src/print_battery_info.c b/src/print_battery_info.c
index 7277765..b2adcb3 100644
--- a/src/print_battery_info.c
+++ b/src/print_battery_info.c
@@ -89,7 +89,7 @@ void print_battery_info(yajl_gen json_gen, char *buffer, int number, const char
                         remaining = atoi(walk+1);
                 }
                 else if (BEGINS_WITH(last, "POWER_SUPPLY_CURRENT_NOW"))
-                        present_rate = atoi(walk+1);
+                        present_rate = abs(atoi(walk+1));
                 else if (BEGINS_WITH(last, "POWER_SUPPLY_VOLTAGE_NOW"))
                         voltage = atoi(walk+1);
                 /* on some systems POWER_SUPPLY_POWER_NOW does not exist, but actually
