---
 src/layout.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/layout.c b/src/layout.c
index c8a21bf..f26fc2d 100644
--- a/src/layout.c
+++ b/src/layout.c
@@ -352,6 +352,7 @@ void render_container(xcb_connection_t *conn, Container *container) {
                 num_clients++;
 
         if (container->mode == MODE_DEFAULT) {
+                int y = 0;
                 int height = (container->height / max(1, num_clients));
                 int rest_pixels = (container->height % max(1, num_clients));
                 DLOG("height per client = %d, rest = %d\n", height, rest_pixels);
@@ -367,15 +368,14 @@ void render_container(xcb_connection_t *conn, Container *container) {
                          * pixel to each client as long as possible. */
                         int this_height = height;
                         if (rest_pixels > 0) {
-                                height++;
+                                this_height++;
                                 rest_pixels--;
                         }
                         /* Check if we changed client->x or client->y by updating it.
                          * Note the bitwise OR instead of logical OR to force evaluation of both statements */
                         if (client->force_reconfigure |
                             update_if_necessary(&(client->rect.x), container->x) |
-                            update_if_necessary(&(client->rect.y), container->y +
-                                        (container->height / num_clients) * current_client) |
+                            update_if_necessary(&(client->rect.y), container->y + y) |
                             update_if_necessary(&(client->rect.width), container->width) |
                             update_if_necessary(&(client->rect.height), this_height))
                                 resize_client(conn, client);
-- 
1.7.5.2

