From f275101a238cf5d6aaf4a20fa039178e8ae8ad30 Mon Sep 17 00:00:00 2001
From: Miguel de Val-Borro <miguel.deval@gmail.com>
Date: Fri, 23 Dec 2011 16:29:26 +0100
Subject: [PATCH] Fix a few spelling typos

---
 docs/userguide |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/docs/userguide b/docs/userguide
index be8335e..f6ed921 100644
--- a/docs/userguide
+++ b/docs/userguide
@@ -174,7 +174,7 @@ Floating windows are always on top of tiling windows.
 i3 stores all information about the X11 outputs, workspaces and layout of the
 windows on them in a tree. The root node is the X11 root window, followed by
 the X11 outputs, then dock areas and a content container, then workspaces and
-finally the windows themselve. In previous versions of i3 we had multiple lists
+finally the windows themselves. In previous versions of i3 we had multiple lists
 (of outputs, workspaces) and a table for each workspace. That approach turned
 out to be complicated to use (snapping), understand and implement.
 
@@ -619,7 +619,7 @@ Only clients that do not cover the whole area of this window expose the color
 used to paint it. If you use a color other than black for your terminals, you
 most likely want to set the client background color to the same color as your
 terminal program's background color to avoid black gaps between the rendered
-area of the termianal and the i3 border.
+area of the terminal and the i3 border.
 
 Colors are in HTML hex format (#rrggbb), see the following example:
 
@@ -1185,7 +1185,7 @@ number or name of the workspace. To move containers to specific workspaces, use
 You can also switch to the next and previous workspace with the commands
 +workspace next+ and +workspace prev+, which is handy, for example, if you have
 workspace 1, 3, 4 and 9 and you want to cycle through them with a single key
-combination. Similarily, you can use +move workspace next+ and +move workspace
+combination. Similarly, you can use +move workspace next+ and +move workspace
 prev+ to move a container to the next/previous workspace.
 
 [[back_and_forth]]
@@ -1195,7 +1195,7 @@ back_and_forth+.
 To move a container to another xrandr output such as +LVDS1+ or +VGA1+, you can
 use the +move output+ command followed by the name of the target output. You
 may also use +left+, +right+, +up+, +down+ instead of the xrandr output name to
-move to the the next output in the specified direction.
+move to the next output in the specified direction.
 
 *Examples*:
 -------------------------
-- 
1.7.3.4

