From c6ef96e5d4f773644308f006d42295ff0aa98851 Mon Sep 17 00:00:00 2001
From: Joep van Delft <mekanik@xs4all.nl>
Date: Thu, 20 Dec 2012 18:32:11 +0100
Subject: [PATCH] Improve tab separators, fixes #894.

---
 src/x.c | 22 +++++++++++++++++-----
 1 file changed, 17 insertions(+), 5 deletions(-)

diff --git a/src/x.c b/src/x.c
index 8f0ae37..3429a05 100644
--- a/src/x.c
+++ b/src/x.c
@@ -465,7 +465,7 @@ void x_draw_decoration(Con *con) {
     xcb_rectangle_t drect = { con->deco_rect.x, con->deco_rect.y, con->deco_rect.width, con->deco_rect.height };
     xcb_poly_fill_rectangle(conn, parent->pixmap, parent->pm_gc, 1, &drect);
 
-    /* 5: draw two unconnected lines in border color */
+    /* 5: draw two unconnected horizontal lines in border color */
     xcb_change_gc(conn, parent->pm_gc, XCB_GC_FOREGROUND, (uint32_t[]){ p->color->border });
     Rect *dr = &(con->deco_rect);
     int deco_diff_l = 2;
@@ -541,19 +541,31 @@ after_title:
 
     /* Draw a separator line after every tab (except the last one), so that
      * tabs can be easily distinguished. */
-    if (parent->layout == L_TABBED && TAILQ_NEXT(con, nodes) != NULL) {
+
+    if (parent->layout == L_TABBED) {
         xcb_change_gc(conn, parent->pm_gc, XCB_GC_FOREGROUND, (uint32_t[]){ p->color->border });
     } else {
         xcb_change_gc(conn, parent->pm_gc, XCB_GC_FOREGROUND, (uint32_t[]){ p->color->background });
     }
     xcb_poly_line(conn, XCB_COORD_MODE_ORIGIN, parent->pixmap, parent->pm_gc, 4,
                   (xcb_point_t[]){
+                      { dr->x + dr->width, dr->y },
+                      { dr->x + dr->width, dr->y + dr->height },
                       { dr->x + dr->width - 1, dr->y },
-                      { dr->x + dr->width - 1, dr->y + dr->height },
-                      { dr->x + dr->width - 2, dr->y },
-                      { dr->x + dr->width - 2, dr->y + dr->height }
+                      { dr->x + dr->width - 1, dr->y + dr->height }
                   });
 
+    /* Draw separator line on left border for focused windows */
+    if (con == focused || con == TAILQ_FIRST(&(parent->focus_head))){
+        xcb_poly_line(conn, XCB_COORD_MODE_ORIGIN, parent->pixmap, parent->pm_gc, 2,
+                      (xcb_point_t[]){
+                          { dr->x, dr->y },
+                          { dr->x, dr->y + dr->height }
+                    });
+    }
+   
+
+
     xcb_change_gc(conn, parent->pm_gc, XCB_GC_FOREGROUND, (uint32_t[]){ p->color->border });
     xcb_poly_segment(conn, parent->pixmap, parent->pm_gc, 2, segments);
 
-- 
1.8.0.2

