From 86e27f714251b89532ff8ec30ba5eb353fc6580e Mon Sep 17 00:00:00 2001
From: Marius Muja <mariusm@cs.ubc.ca>
Date: Mon, 24 Dec 2012 16:16:47 -0800
Subject: [PATCH] Handle desktop type windows

---
 include/atoms.xmacro |    1 +
 src/manage.c         |    8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/include/atoms.xmacro b/include/atoms.xmacro
index af60b96..d841afc 100644
--- a/include/atoms.xmacro
+++ b/include/atoms.xmacro
@@ -5,6 +5,7 @@ xmacro(_NET_WM_STATE_FULLSCREEN)
 xmacro(_NET_WM_STATE)
 xmacro(_NET_WM_WINDOW_TYPE)
 xmacro(_NET_WM_WINDOW_TYPE_DOCK)
+xmacro(_NET_WM_WINDOW_TYPE_DESKTOP)
 xmacro(_NET_WM_WINDOW_TYPE_DIALOG)
 xmacro(_NET_WM_WINDOW_TYPE_UTILITY)
 xmacro(_NET_WM_WINDOW_TYPE_TOOLBAR)
diff --git a/src/manage.c b/src/manage.c
index 9835aa2..8f4250b 100644
--- a/src/manage.c
+++ b/src/manage.c
@@ -202,6 +202,14 @@ void manage_window(xcb_window_t window, xcb_get_window_attributes_cookie_t cooki
     Con *search_at = croot;
 
     xcb_get_property_reply_t *reply = xcb_get_property_reply(conn, wm_type_cookie, NULL);
+
+    /* Don't manage desktop type windows, just map them */
+    if (xcb_reply_contains_atom(reply, A__NET_WM_WINDOW_TYPE_DESKTOP)) {
+        xcb_map_window(conn, window);
+        LOG("Ignoring window of type desktop\n");
+        goto geom_out;
+    }
+
     if (xcb_reply_contains_atom(reply, A__NET_WM_WINDOW_TYPE_DOCK)) {
         LOG("This window is of type dock\n");
         Output *output = get_output_containing(geom->x, geom->y);
-- 
1.7.10.4

