--- i3-4.1/docs/userguide	2011-11-11 23:41:26.000000000 +0100
+++ i3-4.1.new/docs/userguide	2011-12-17 21:48:19.983926316 +0100
@@ -639,20 +639,20 @@ the window.
 === Interprocess communication
 
 i3 uses unix sockets to provide an IPC interface. This allows third-party
-programs to get information from i3, such as the current workspaces
-(to display a workspace bar), and to control i3.
+programs to get information from i3, such as the current workspaces (to display
+a workspace bar), and to control i3.
 
 The IPC socket is enabled by default and will be created in
-+/tmp/i3-%u/ipc-socket.%p+ where +%u+ is your UNIX username and +%p+ is the PID
-of i3.
++/tmp/i3-%u.XXXXXX/ipc-socket.%p+ where +%u+ is your UNIX username and +%p+ is
+the PID of i3 and XXXXXX is a string of random alpha characters.
 
 You can override the default path through the environment-variable +I3SOCK+ or
-by specifying the +ipc-socket+ directive. This is discouraged, though, since i3
-does the right thing by default.
+by specifying the +ipc_socket+ directive. If you decide to change it it is
+strongly recommended to set this to a location in your home directory.
 
 *Examples*:
 ----------------------------
-ipc-socket /tmp/i3-ipc.sock
+ipc_socket ~/.i3/socket
 ----------------------------
 
 You can then use the +i3-msg+ application to perform any command listed in
