From d324de0a3cdda21879dc669093f98591aab5216f Mon Sep 17 00:00:00 2001
From: mrvmyr <mrvmyr@gmail.com>
Date: Fri, 18 Jul 2014 22:08:47 +0200
Subject: [PATCH] focus left/right for floating containers raises as well

---
 src/tree.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/tree.c b/src/tree.c
index da73e14..4c9c440 100644
--- a/src/tree.c
+++ b/src/tree.c
@@ -591,7 +591,7 @@ static bool _tree_next(Con *con, char way, orientation_t orientation, bool wrap)
     Con *parent = con->parent;

     if (con->type == CT_FLOATING_CON) {
-        /* left/right focuses the previous/next floating container */
+        /* left/right focuses and raises the previous/next floating container */
         if (orientation == HORIZ) {
             Con *next;
             if (way == 'n')
@@ -612,6 +612,7 @@ static bool _tree_next(Con *con, char way, orientation_t orientation, bool wrap)
                 return false;

             con_focus(con_descend_focused(next));
+            floating_raise_con(next);
             return true;
         } else {
             /* up/down cycles through the Z-index */
--
2.0.1

