From: Julian Ospald <hasufell@gentoo.org>
Date: Tue Apr 22 15:09:23 UTC 2014
Subject: respect CFLAGS in linking command

--- i3/src/i3.mk
+++ i3/src/i3.mk
@@ -66,7 +66,7 @@
 
 i3: libi3.a $(i3_OBJECTS)
 	echo "[i3] Link i3"
-	$(CC) $(I3_LDFLAGS) $(LDFLAGS) -o $@ $(filter-out libi3.a,$^) $(LIBS) $(i3_LIBS)
+	$(CC) $(CFLAGS) $(I3_LDFLAGS) $(LDFLAGS) -o $@ $(filter-out libi3.a,$^) $(LIBS) $(i3_LIBS)
 
 install-i3: i3
 	echo "[i3] Install"
