--- a/src/workspace.c	2012-11-11 14:45:23.566224130 +0100
+++ b/src/workspace.c	2012-11-11 15:04:14.707923122 +0100
@@ -343,7 +343,7 @@
      * the corresponding workspace is cleaned up.
      * NOTE: Internal cons such as __i3_scratch (when a scratchpad window is
      * focused) are skipped, see bug #868. */
-    if (current && !con_is_internal(current)) {
+    if (current && !(current->name[0] == '_' && current->name[1] == '_')) {
         FREE(previous_workspace_name);
         if (current) {
             previous_workspace_name = sstrdup(current->name);
