diff --git i/libi3/ipc_send_message.c w/libi3/ipc_send_message.c
index 850fbdd..781ff59 100644
--- i/libi3/ipc_send_message.c
+++ w/libi3/ipc_send_message.c
@@ -38,14 +38,12 @@ int ipc_send_message(int sockfd, uint32_t message_size,
     memcpy(walk, payload, message_size);
 
     int sent_bytes = 0;
-    int bytes_to_go = buffer_size;
-    while (sent_bytes < bytes_to_go) {
-        int n = write(sockfd, msg + sent_bytes, bytes_to_go);
+    while (sent_bytes < buffer_size) {
+        int n = write(sockfd, msg + sent_bytes, buffer_size - sent_bytes);
         if (n == -1)
             return -1;
 
         sent_bytes += n;
-        bytes_to_go -= n;
     }
 
     return 0;
