From c2158f63f1bc486833c3274801b7907e2ad58d6e Mon Sep 17 00:00:00 2001
From: beauby <lucas.hosseini@gmail.com>
Date: Thu, 9 Feb 2012 01:47:18 +0100
Subject: [PATCH 2/3] BugFix: keep the indentation of config.keycodes

---
 i3-config-wizard/main.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/i3-config-wizard/main.c b/i3-config-wizard/main.c
index ee1ab33..244f463 100644
--- a/i3-config-wizard/main.c
+++ b/i3-config-wizard/main.c
@@ -342,8 +342,11 @@ static void finish() {
 
         /* Skip leading whitespace */
         char *walk = line;
-        while (isspace(*walk) && walk < (line + len))
+        while (isspace(*walk) && walk < (line + len)) {
+            /* Pre-output the skipped whitespaces to keep proper indentation */
+            fputc(*walk, ks_config);
             walk++;
+        }
 
         if (strncmp(walk, "set $mod ", strlen("set $mod ")) == 0) {
             /* Set the modifier the user chose */
-- 
1.7.9

