From e6a3980cb848b3516b57d8ecd0b3172300db96d8 Mon Sep 17 00:00:00 2001
From: beauby <lucas.hosseini@gmail.com>
Date: Fri, 10 Feb 2012 00:32:55 +0100
Subject: [PATCH 3/3] Fixed a memory leak

---
 i3-config-wizard/main.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/i3-config-wizard/main.c b/i3-config-wizard/main.c
index 352fee5..9c3fbe7 100644
--- a/i3-config-wizard/main.c
+++ b/i3-config-wizard/main.c
@@ -291,6 +291,7 @@ static void finish() {
     FILE *ks_config = fopen(config_path, "w");
     if (ks_config == NULL)
         err(1, "Could not open output config file \"%s\"", config_path);
+    free(config_path);
 
     char *line = NULL;
     size_t len = 0;
-- 
1.7.9

