From 12c556da09fc9a81fd1c8184035a8e28b7a2527f Mon Sep 17 00:00:00 2001
From: Miguel de Val-Borro <miguel.deval@gmail.com>
Date: Mon, 26 Dec 2011 17:41:29 +0100
Subject: [PATCH] Update section 8.1 to describe the configuration of i3bar and i3status in v4

---
 docs/userguide |   21 +++++++++++++++++----
 1 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/docs/userguide b/docs/userguide
index e75f1ca..9b5c5e4 100644
--- a/docs/userguide
+++ b/docs/userguide
@@ -840,6 +840,7 @@ mode <dock|hide>
 mode hide
 ----------------
 
+[[i3bar_position]]
 === Position
 
 This option determines in which edge of the screen i3bar should show up.
@@ -1534,14 +1535,26 @@ If you don’t already have your favorite way of generating such a status line
 this task. It was written in C with the goal of using as few syscalls as
 possible to reduce the time your CPU is woken up from sleep states. Because
 i3status only spits out text, you need to combine it with some other tool, like
-i3bar. Use a pipe to connect them: +i3status | i3bar -d+.
+i3bar, opening a +bar+ block in the config file.
+
+---------------------------
+bar {
+    status_command i3status
+}
+---------------------------
 
 Regardless of which application you use to display the status line, you
 want to make sure that it registers as a dock window using EWMH hints. i3 will
 position the window either at the top or at the bottom of the screen, depending
-on which hint the application sets. With i3bar, you can use +-d+ or +-dbottom+
-for positioning it at the bottom and +-dtop+ to position it at the top of the
-screen.
+on which hint the application sets. With i3bar, the position option determines
+in which edge of the screen the bar will be drawn (see <<i3bar_position>>).
+
+---------------------------
+bar {
+    position bottom
+    status_command i3status
+}
+---------------------------
 
 === Giving presentations (multi-monitor)
 
-- 
1.7.3.4

