From 7a0c6e4032e2f8d95ef84abefaf0138373aa3b23 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fernando=20Tarl=C3=A1=20Cardoso=20Lemos?= <fernandotcl@gmail.com>
Date: Sat, 5 Feb 2011 14:23:02 -0200
Subject: [PATCH] Fix libxcursor -> xcb cursors fallback.

---
 src/xcursor.c |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/xcursor.c b/src/xcursor.c
index cd80aa6..ee77d0c 100644
--- a/src/xcursor.c
+++ b/src/xcursor.c
@@ -17,17 +17,17 @@ static const int xcb_cursors[XCURSOR_CURSOR_MAX] = {
     XCB_CURSOR_SB_V_DOUBLE_ARROW
 };
 
-static Cursor load_cursor(const char *name, int font) {
+static Cursor load_cursor(const char *name) {
     Cursor c = XcursorLibraryLoadCursor(xlibdpy, name);
     if (c == None)
-        c = XCreateFontCursor(xlibdpy, font);
+        xcursor_supported = false;
     return c;
 }
 
 void xcursor_load_cursors() {
-    cursors[XCURSOR_CURSOR_POINTER] = load_cursor("left_ptr", XC_left_ptr);
-    cursors[XCURSOR_CURSOR_RESIZE_HORIZONTAL] = load_cursor("sb_h_double_arrow", XC_sb_h_double_arrow);
-    cursors[XCURSOR_CURSOR_RESIZE_VERTICAL] = load_cursor("sb_v_double_arrow", XC_sb_v_double_arrow);
+    cursors[XCURSOR_CURSOR_POINTER] = load_cursor("left_ptr");
+    cursors[XCURSOR_CURSOR_RESIZE_HORIZONTAL] = load_cursor("sb_h_double_arrow");
+    cursors[XCURSOR_CURSOR_RESIZE_VERTICAL] = load_cursor("sb_v_double_arrow");
 }
 
 Cursor xcursor_get_cursor(enum xcursor_cursor_t c) {
-- 
1.7.2.3

