diff --git a/i3lock.c b/i3lock.c
index fffb3da..11b6c0a 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -42,6 +42,7 @@ xcb_window_t win;
 static xcb_cursor_t cursor;
 static pam_handle_t *pam_handle;
 int input_position = 0;
+bool passwordless = false;
 /* Holds the password you enter (in UTF-8). */
 static char password[512];
 static bool beep = false;
@@ -173,7 +174,7 @@ static void clear_pam_wrong(EV_P_ ev_timer *w, int revents) {
 }
 
 static void input_done(void) {
-    if (input_position == 0)
+    if (! passwordless && input_position == 0 )
         return;
 
     if (clear_pam_wrong_timeout) {
@@ -251,6 +252,7 @@ static void handle_key_press(xcb_key_press_event_t *event) {
     n = xkb_keysym_to_utf8(ksym, buffer, sizeof(buffer));
 
     switch (ksym) {
+    case XKB_KEY_XF86ScreenSaver:
     case XKB_KEY_Return:
     case XKB_KEY_KP_Enter:
         password[input_position] = '\0';
@@ -512,6 +514,7 @@ int main(int argc, char *argv[]) {
         {"debug", no_argument, NULL, 0},
         {"help", no_argument, NULL, 'h'},
         {"no-unlock-indicator", no_argument, NULL, 'u'},
+        {"passwordless", no_argument, NULL, 'l'},
         {"image", required_argument, NULL, 'i'},
         {"tiling", no_argument, NULL, 't'},
         {NULL, no_argument, NULL, 0}
@@ -520,7 +523,7 @@ int main(int argc, char *argv[]) {
     if ((username = getenv("USER")) == NULL)
         errx(1, "USER environment variable not set, please set it.\n");
 
-    while ((o = getopt_long(argc, argv, "hvnbdc:p:ui:t", longopts, &optind)) != -1) {
+    while ((o = getopt_long(argc, argv, "hvnbdc:p:uli:t", longopts, &optind)) != -1) {
         switch (o) {
         case 'v':
             errx(EXIT_SUCCESS, "version " VERSION " © 2010-2012 Michael Stapelberg");
@@ -563,6 +566,9 @@ int main(int argc, char *argv[]) {
                 errx(1, "i3lock: Invalid pointer type given. Expected one of \"win\" or \"default\".\n");
             }
             break;
+        case 'l':
+            passwordless = true;
+            break;
         case 0:
             if (strcmp(longopts[optind].name, "debug") == 0)
                 debug_mode = true;
