diff --git i/src/render.c w/src/render.c
index 9c0debb..16fd751 100644
--- i/src/render.c
+++ w/src/render.c
@@ -111,6 +111,7 @@ void render_con(Con *con, bool render_fullscreen) {
     DLOG("Rendering %snode %p / %s / layout %d / children %d\n",
          (render_fullscreen ? "fullscreen " : ""), con, con->name, con->layout,
          children);
+    DLOG("FYI: node %d,%d,%dx%d\n", con->rect.x, con->rect.y, con->rect.width, con->rect.height);
 
     /* Copy container rect, subtract container border */
     /* This is the actually usable space inside this container for clients */
@@ -346,13 +347,15 @@ void render_con(Con *con, bool render_fullscreen) {
             child->rect.height = rect.height;
 
             child->deco_rect.width = child->rect.width / children;
-            child->deco_rect.height = deco_height;
             child->deco_rect.x = x - con->rect.x + i * child->deco_rect.width;
             child->deco_rect.y = y - con->rect.y;
 
             if (children > 1 || (child->border_style != BS_1PIXEL && child->border_style != BS_NONE)) {
                 child->rect.y += deco_height;
                 child->rect.height -= deco_height;
+                child->deco_rect.height = deco_height;
+            } else {
+                child->deco_rect.height = (child->border_style == BS_1PIXEL ? 1 : 0);
             }
         }
 
@@ -370,8 +373,10 @@ void render_con(Con *con, bool render_fullscreen) {
             y += child->rect.height;
         }
 
-        DLOG("child at (%d, %d) with (%d x %d)\n",
-                child->rect.x, child->rect.y, child->rect.width, child->rect.height);
+        DLOG("child at (%d, %d) with (%d x %d), deco_rect (%d,%d,%dx%d)\n",
+                child->rect.x, child->rect.y, child->rect.width, child->rect.height,
+                child->deco_rect.x, child->deco_rect.y, child->deco_rect.width,
+                child->deco_rect.height);
         x_raise_con(child);
         render_con(child, false);
         i++;
